drivers/adc/adc_shell.c

```diff
@@ -476,13 +453,13 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_adc_cmds,
 );
 
 SHELL_STATIC_SUBCMD_SET_CREATE(sub_adc,
-#if DT_HAS_DRV_INST(0)
+#if DT_HAS_NODE_STATUS_OKAY(DT_DRV_INST(0))
```

Comment:
I don't think this is correct.  if we have inst0 - ok, inst1 - disabled, inst2-ok, inst3-ok.  The DT_INST_FOREACH_OKAY will get inst0, inst2, inst3.  But dont here we aren't going to match.

Reply:
changed to:

```
#define ADC_SHELL_COMMAND(inst) \
	SHELL_CMD(ADC_##inst, &sub_adc_cmds, "ADC_" #inst, NULL),

SHELL_STATIC_SUBCMD_SET_CREATE(
	sub_adc,
	DT_INST_FOREACH_OKAY(ADC_SHELL_COMMAND)
	SHELL_SUBCMD_SET_END /* Array terminated. */
);
```